Neos.Neos/Classes/NodeTypePostprocessor/CreationDialogPostprocessor.php

```diff
@@ -0,0 +1,102 @@
+<?php
+namespace Neos\Neos\NodeTypePostprocessor;
+
+/*
+ * This file is part of the Neos.Neos package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\ContentRepository\Domain\Model\NodeType;
+use Neos\ContentRepository\NodeTypePostprocessor\NodeTypePostprocessorInterface;
+use Neos\Utility\Arrays;
+use Neos\Utility\PositionalArraySorter;
+
+/**
+ * Node Type post processor that looks for properties flagged with "showInCreationDialog" and sets the "creationDialog" configuration accordingly
+ *
+ * Example NodeTypes.yaml configuration:
+ *
+ * 'Some.Node:Type':
+ *   # ...
+ *   properties:
+ *     'someProperty':
+ *       type: string
+ *       ui:
+ *         label: 'Link'
+ *         showInCreationDialog: true
+ *         inspector:
+ *           editor: 'Neos.Neos/Inspector/Editors/LinkEditor'
+ *
+ * Will be converted to:
+ *
+ * 'Some.Node:Type':
+ *   # ...
+ *   ui:
+ *     creationDialog:
+ *       elements:
+ *         'someProperty':
+ *           type: string
+ *           ui:
+ *             label: 'Link'
+ *             editor: 'Neos.Neos/Inspector/Editors/LinkEditor'
+ *   properties:
+ *     'someProperty':
+ *       # ...
+ */
+class CreationDialogPostprocessor implements NodeTypePostprocessorInterface
+{
+    /**
+     * @param NodeType $nodeType (uninitialized) The node type to process
+     * @param array $configuration input configuration
+     * @param array $options The processor options
+     * @return void
+     */
+    public function process(NodeType $nodeType, array &$configuration, array $options): void
+    {
+        if (!isset($configuration['properties'])) {
+            return;
+        }
+        $creationDialogElements = [];
+        foreach ($configuration['properties'] as $propertyName => $propertyConfiguration) {
+            if (!isset($propertyConfiguration['ui']['showInCreationDialog']) || $propertyConfiguration['ui']['showInCreationDialog'] !== true) {
+                continue;
+            }
+            $creationDialogElement = $this->convertPropertyConfiguration($nodeType->getConfiguration('properties.' . $propertyName) ?? []);
```

Comment:
For some weird reason we amend the node configuration via an AOP aspect upon creation of the `NodeType` instance.. Therefore I use that configuration as basis here so I don't have to duplicate a whole lot of logic for automagic i18n and editor configuration